boltdesignsystem/bolt
# DS-23 | Convert typography and spacing scss vars to CSS custom props

## Jira

https://pegadigitalit.atlassian.net/browse/DS-23

## Summary

Added new CSS custom props for spacing and typography.

## Details

1. Added new spacing maps in core settings
2. Added new typography maps in core settings
3. Generated custom CSS props based on the new maps in global
4. Converted Text and Grid components to use the CSS custom props
5. Added warnings for old spacing functions and mixins
6. Added warnings for old font functions and mixins

## How to test

Run the branch locally and do the following:

- Review Text and Grid component docs
- Find a scss file using the old `bolt-margin` and `bolt-padding` mixin/function, change something and save it, make sure warnings show up in terminal
- Find a scss file using the old `bolt-font-family`, `bolt-font-weight`, and `bolt-font-size` mixin/function, change something and save it, make sure warnings show up in terminal
- Review the spacing-vars and typography-vars scss files in global/generic
